pkg/diff/merge.go

```diff
@@ -47,35 +45,39 @@ func (r RowDatas) Less(i, j int) bool {
 		if !ok {
 			log.Fatal("data don't have column", zap.String("column", col.Name.O), zap.Reflect("data", r.Rows[j].Data))
 		}
-		data1 = col1.Data
-		data2 = col2.Data
 
-		if needQuotes(col.FieldType) {
-			strData1 := string(data1)
-			strData2 := string(data2)
+		strData1 := string(col1.Data)
+		strData2 := string(col2.Data)
 
+		if col1.IsNull {
+			if col2.IsNull {
+				continue
+			}
+
+			return true
+		}
+
+		if col2.IsNull {
+			return false
+		}
+
+		if needQuotes(col.FieldType) {
 			if strData1 == strData2 {
-				// `NULL` is less than ""
-				if r.Rows[i].Data[col.Name.O].IsNull {
-					return true
-				}
-				if r.Rows[j].Data[col.Name.O].IsNull {
-					return false
-				}
 				continue
 			}
 			if strData1 > strData2 {
 				return false
 			}
 			return true
 		}
-		num1, err1 := strconv.ParseFloat(string(data1), 64)
+
+		num1, err1 := strconv.ParseFloat(strData1, 64)
 		if err1 != nil {
-			log.Fatal("convert string to float failed", zap.ByteString("data", data1), zap.Error(err1))
+			log.Fatal("convert string to float failed", zap.String("data", strData1), zap.Error(err1))
```

Comment:
log the column name for debugging?

Reply:
done https://github.com/pingcap/tidb-tools/pull/250/commits/d0ed8ba163809d2efa86718946636e59e3d1c5e7